src/main/kotlin/com/hibob/academy/employeeFeedback/resource/FeedbackResource.kt

```diff
@@ -0,0 +1,58 @@
+package com.hibob.academy.employeeFeedback.resource
+
+import com.hibob.academy.employeeFeedback.dao.AnonymityType.Companion.convertStringToAnonymityType
+import com.hibob.academy.employeeFeedback.dao.FeedbackSubmitRequest
+import com.hibob.academy.employeeFeedback.service.AuthenticatedUsersService
+import jakarta.ws.rs.core.Response
+import com.hibob.academy.employeeFeedback.service.FeedbackService
+import jakarta.ws.rs.*
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.core.Context
+import jakarta.ws.rs.core.MediaType
+import org.springframework.stereotype.Controller
+import org.springframework.web.bind.annotation.RequestBody
+
+
+@Controller
+@Path("/api/feedback")
+@Produces(MediaType.APPLICATION_JSON)
+class FeedbackResource(private val feedbackService: FeedbackService, private val authenticatedUsersService: AuthenticatedUsersService) {
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun submitFeedback(@RequestBody feedbackRequest: FeedbackSubmitRequest, @Context request: ContainerRequestContext): Response {
+        val employeeId = authenticatedUsersService.getLoggedInEmployeeId(request)
+        val companyId = authenticatedUsersService.getLoggedInCompanyId(request)
+        val anonymity = convertStringToAnonymityType(feedbackRequest.anonymity)
+
+        val success = feedbackService.submitFeedback(employeeId, companyId, anonymity, feedbackRequest.feedback)
+
+        if (success) return Response.ok("Feedback created Successfully").build()
+
+        return Response.status(Response.Status.BAD_REQUEST).entity("Unable to create feedback").build()
+    }
+
+    @GET
+    @Path("/allFeedback")
+    fun viewFeedback(@Context request: ContainerRequestContext): Response {
+        val employeeId = authenticatedUsersService.getLoggedInEmployeeId(request)
+        val companyId = authenticatedUsersService.getLoggedInCompanyId(request)
+        val role = authenticatedUsersService.getLoggedInRole(request)
+
+        if (!authenticatedUsersService.validateRole(role)) return Response.status(Response.Status.UNAUTHORIZED).build()
```

Comment:
you don't have to change it, but so you know, there are 2 options here.
What you did - check, return value, validate the value
and the other option is that the authentication service will throw the exception, you won't catch it here and it will bubble up.

It is called "asking permission vs asking for forgiveness" (the second one is because you throw an error :) )

Reply:
I didn't really understand